### PR TITLE
fix: do not handle all !=TRANSFER transactions as VOTE

### PR DESCRIPTION
**Pull request by vasild**
## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [ ] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by vasild** [comment]
Notice: this patch will most likely break things as it would invalidate any non-TRANSFER and non-VOTE transaction. I guess the invalidation added is too strict and should be relaxed.

File: packages/core-transaction-pool/lib/guard.js [hunk 1]
```diff
@@ -165,15 +165,28 @@ module.exports = class TransactionGuard {
...
-      if (this.pool.checkIfSenderHasVoteTransactions(transaction.senderPublicKey)) {
-        this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending vote transaction in pool`)
+        break
+      case TRANSACTION_TYPES.VOTE:
+        if (this.pool.checkIfSenderHasVoteTransactions(transaction.senderPublicKey)) {
+          this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending vote transaction in pool`)
+          return
+        }
+        break
+      case TRANSACTION_TYPES.SECOND_SIGNATURE:
```
Comment (faustbrian) [inline packages/core-transaction-pool/lib/guard.js]: `default` should be sufficient if none of the other types are handled in another case.

Comment (vasild) [reply packages/core-transaction-pool/lib/guard.js]: Yes, technically `default:` suffices, but I added them in order to be explicit at what is being invalidated. I presume nobody remembers all transaction types on top of his head, so reading them here helps to skip looking up the definition of `TRANSACTION_TYPES`.

Comment (dated) [inline packages/core-transaction-pool/lib/guard.js]:
```suggestion
      case TRANSACTION_TYPES.SECOND_SIGNATURE:
        if (this.pool.checkIfSenderHasSecondSignatureTransactions(transaction.senderPublicKey)) {
          this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending second signature transaction in pool`)
          return
        }
        break
```

File: packages/core-transaction-pool/lib/guard.js [hunk 2]
```diff
@@ -165,15 +165,28 @@ module.exports = class TransactionGuard {
...
+      case TRANSACTION_TYPES.DELEGATE_REGISTRATION:
```
Comment (dated) [inline packages/core-transaction-pool/lib/guard.js]:
```suggestion
      case TRANSACTION_TYPES.DELEGATE_REGISTRATION:
        if (this.pool.checkIfSenderHasDelegateRegistrationTransactions(transaction.senderPublicKey)) {
          this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending delegate registration transaction in pool`)
          return
        }
        break
```

**Pull request merged**